apps/src/code-studio/pd/foorm/editor/form/FoormFormSaveBar.jsx

```diff
@@ -61,10 +63,12 @@ class FoormFormSaveBar extends Component {
 
   publishUrl = () => `/foorm/forms/${this.props.formId}/publish`;
 
-  handleSave = () => {
+  handleSave = (saveAsNewVersion = false) => {
     this.setState({isSaving: true});
     this.props.setSaveError(null);
-    if (this.props.isFormPublished) {
+    if (saveAsNewVersion) {
+      this.saveNewForm(true);
```

Comment:
Am I reading this right that we'd save a new version with no warning if you click the "Save as New Version" button? That makes me a little nervous, as I think in most (all?) cases we show the most recent version of a given Form.

Reply:
Yeah I think it makes sense to add a warning popup. We tie the url to a specific version though, right? Where would we be automatically using the latest version of a form? Maybe with libraries..

Reply:
I think we in most (all?) cases default to using the latest version of a Form, like what we were looking at yesterday with the daily survey stuff:

https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/controllers/pd/workshop_daily_survey_controller.rb#L260

I was also thinking that we probably should not show this button if you're looking at a version of Form that is not the most recent version.

Reply:
ooh got it. so the simple survey ones are definitely tied to a version, but that makes sense for these ones. also makes sense to not show the button if there is a more recent version... i'll have to figure out a nice way to determine that.

Reply:
Ah right, good call on simple surveys. I think I wavered on whether to require you to specify a version or not and guess I ended up requiring it :).

Seems like the logic to show only most recent version in dropdown menu and this could be shared? I dunno if Lodash isn't fashionable any more but I feel like some sort of groupBy/maxBy thing could work

Reply:
So we're going no warning on saving a new version? Maybe check with Tess?

Reply:
Oh, added that in an earlier commit here: https://github.com/code-dot-org/code-dot-org/pull/40734/commits/1e5b20653e13d2051485d14cad752d1f661fc940